### PR TITLE
Longevity enhancements

### PR DESCRIPTION
**Pull request by PrasadDesala**
This PR covers the following,
- [x] Include running MCG features in Longevity Stage 2 runs (setting up MCG features buckets, run IOs, validate IOs)
- [x] Stage2() can now be called to run mcg and non-mcg operations individually 
- [x] Fixes: #10002 
- [x] Cluster sanity checks can be collected at the start of the cycle and at the end of each iteration for stage 2, 3 and 4

### CODE REVIEW

File: tests/cross_functional/conftest.py [hunk 1]
```diff
@@ -1346,6 +1346,6 @@ def factory(
         )
         feature_setup_map["executor"]["event"] = event
         feature_setup_map["executor"]["threads"] = threads
-        return feature_setup_map
+        return feature_setup_map, all_buckets
```
Comment (mashetty330) [inline tests/cross_functional/conftest.py:1349]: this change will break the existing system tests @PrasadDesala because of unpacking.

Comment (mashetty330) [reply tests/cross_functional/conftest.py:1349]: instead you can add all_buckets to the feature_setup_map dict with key "all_buckets"

**Review by mashetty330 — CHANGES_REQUESTED** [review]

File: tests/cross_functional/conftest.py (hunk 1 above)
Comment (PrasadDesala) [reply tests/cross_functional/conftest.py]: Done

**Review by mashetty330 — APPROVED** [review]

**Comment by PrasadDesala** [comment]
verification job - https://url.corp.redhat.com/stag2-verification

**Review by vavuthu — APPROVED** [review]

**Pull request merged**